crates/wasi-provider/src/states/initializing.rs

```diff
@@ -118,20 +118,25 @@ impl State<PodState> for Initializing {
 
                         // If we are in a failed state, insert in the init containers we already ran
                         // into a pod handle so they are available for future log fetching
-                        let pod_handle = Handle::new(container_handles, pod.clone(), None).await?;
+                        let pod_handle = Handle::new(container_handles, pod.clone(), None);
                         let pod_key = key_from_pod(&pod);
                         {
                             let mut handles = pod_state.shared.handles.write().await;
                             handles.insert(pod_key, pod_handle);
                         }
-                        client
-                            .patch_status(
-                                pod.name(),
-                                &PatchParams::default(),
-                                serde_json::to_vec(&s)?,
-                            )
-                            .await?;
-                        return Ok(Transition::next(self, Error { message }));
+
+                        let status_json = match serde_json::to_vec(&s) {
+                            Ok(json) => json,
+                            Err(e) => fail_fatal!(e),
+                        };
+
+                        match client
+                            .patch_status(pod.name(), &PatchParams::default(), status_json)
+                            .await
```

Comment:
Should this fail? Or just be logged. A control plane outage doesnt necessarily need to stop the Pod.

Reply:
Yeah, I don't think this should be a fatal fail. Definitely log an error though

Reply:
This was replicating previous behaviour where the `?` bubbled up to exit the run loop.  I can change it though.

Reply:
Yeah, @kflansburg and @thomastaylor312 pointed out a bug here where we shouldn't fail. Would you mind changing this so that we only log the error message, @itowlson? I know it's a little more work here, but it could save us a couple of erroneous error cases in the future.

Reply:
That being said, the same argument applied [in another comment](https://github.com/deislabs/krustlet/pull/398#discussion_r495621934) could apply here as well. In which case, I'd be okay with that and file a ticket to resolve this in a follow-up.